ui/app/adapters/sync/association.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import ApplicationAdapter from 'vault/adapters/application';
+
+export default class SyncDestinationAdapter extends ApplicationAdapter {
+  namespace = 'v1/sys/sync/destinations';
+
+  buildURL(modelName, id, snapshot, requestType, query) {
+    const { destinationType, destinationName, syncStatus } = snapshot ? snapshot.attributes() : query;
+    // use sync_status to determine whether saving a record should use set or remove endpoint
+    // new records will not have a status so the only scenario where remove should be used if status is synced
+    const action = requestType === 'query' ? '' : syncStatus === 'SYNCED' ? '/remove' : '/set';
+    return `${super.buildURL()}/${destinationType}/${destinationName}/associations${action}`;
+  }
```

Comment:
I considered making custom adapter methods to explicitly set or remove an association but decided to use the `sync_status` value (or lack thereof) to determine which endpoint to use.

Reply:
Are we sure there is not a case where a user needs to re-set a synced secret?

Reply:
Yes as far as I can tell a user should be able to set the same association after it was removed. It doesn't look like removing the association deletes it which is why I opted to use `model.save` for both actions. The status changes to `UNSYNCED` according to the docs after it was removed so a user could then go and sync it again.

Reply:
I was more wondering if there was a case where a secret returns `SYNCED` but perhaps something about the destination changes (maybe a credential?) and the user wants to re-set before the status has updated. 

So I guess it's more of a question how reliable the status is and if there's a case a synced secret will need to be re-set.

Reply:
Right and perhaps the user is able to force a sync using the set endpoint too. Rather than waiting to find out I'll update and add custom adapter methods instead which will allow us to be explicit in their use.

Reply:
I ended up utilizing `adapterOptions` to define the action when save is called which seems a bit more streamlined to me than custom adapter methods after looking into it further. I also added an assertion and a test for it to help during development if we miss passing the adapterOptions.